tests/test_backend.py

```diff
@@ -38,3 +38,19 @@ def test_future_env_select(self):
 
         with self.assertRaises(NotImplementedError):
             PyRDF.use("dask")
+
+class BackendInitTest(unittest.TestCase):
+    """
+    Tests to ensure that the Backend abstract
+    class cannot be instantiated.
+
+    """
+    def test_backend_init_error(self):
```

Comment:
@shravan97 We should also test a subclass of `Backend` which does not implement `execute`, since this is the real case we want to avoid. For that you can just create a dummy class on the test itself before the `assert`.